draft-thomson-train-protocol.md

```diff
@@ -99,167 +105,324 @@ cannot assume that rate limit indications from one path apply to new paths.
+  Rate Signal (6),
+  Version (32) = 0xTBD,
+  Destination Connection ID Length (8),
+  Destination Connection ID (0..2040),
+  Source Connection ID Length (8) = 0,
+}
 ~~~
-cfXXXXXXXX0008f067a5502a4262b574 6f6b656ec8646ce8bfe33952d9555436
-65dcc7b6
-~~~
+{: #fig-train-packet title="TRAIN Packet Format"}
 
-But the 4 bytes of the version number will be restored to 0x6b3343cf before
-attempting to process the packet further.
+The most significant bit (0x80) of the packet indicates that this is a QUIC long
+header packet.  The next bit (0x40) is reserved and can be set according to
+{{!QUIC-BIT=RFC9287}}.
 
+The entire payload of the TRAIN packet is carried in the Rate Signal field that
+forms the low 6 bits (0x3f) of the first byte.  Values for this field are
+described in {{rate-signal}}.
 
-# Deployment
+This packet includes a Destination Connection ID field that is set to the same
+value as other packets in the same datagram; see {{Section 12.2 of QUIC}}.
+
+The Source Connection ID Length field is set to zero, meaning that the Source
+Connection ID field is empty.  TRAIN packets can still be sent during the QUIC
+handshake; the Source Connection ID field in other packets might be needed to
+successfully complete a handshake.
+
+TRAIN packets SHOULD be included as the first packet in a datagram.  This is
+necessary in many cases for QUIC versions 1 and 2 because packets with a short
+header cannot precede any other packets.
 
-Rate limiting signals that are applied in this way do not guarantee that the
-signal is successfully received.
 
-The primary cost associated with the application of rate limit signals is the
-potential for the packet to be dropped as a consequence of the changed version.
-An endpoint that does not recognize these new versions will treat the
-modification as damage and discard the packet.
+## Rate Signals {#rate-signal}
 
 {:aside}
-> TODO: do we need to define a new QUIC version where support for that version
-> also indicates support for this feature?  That might allow endpoints to know
-> that their long header packets, if modified, won't get dropped.
+> Note: The exact set of rates that are included is subject to negotiation.
+> We should aim to find typical rate limits that are used in real networks
+> and by real applications.
+
+The Rate Signal field of a TRAIN packet is set to 0xTBD when sent by a QUIC
+endpoint.  Receiving value indicates that there is no rate limit in place or
+that the TRAIN protocol is not supported by network elements on the path.
+
+The values from {{table-rates}} are specified to carry a the corresponding rate
+limit signal.
+
+| Rate Signal | Rate Limit |
+|:------------|:-----------|
+| 0xTBD       | X Mbps     |
+| 0xTBD       | X Mbps     |
+| 0xTBD       | X Mbps     |
+| 0xTBD       | X Mbps     |
+| 0xTBD       | X Mbps     |
+| 0xTBD       | X Mbps     |
+| 0xTBD       | X Mbps     |
+| 0xTBD       | X Mbps     |
+{: #table-rates title="Table of Rate Signals and Rate Limits"
+
```

Comment:
Have we not decided to state some kind of bucket size or enforcement period as well?